### ISSUE TITLE
Extend ~ tilde support

### ISSUE THREAD

**Issue by imperugo** [aspnet/Razor#223]
Hi, 
could be very helpful if the ~ could have more power than a simple replace from ~/Styles/main.css to /Styles/main.css or /MyVirtual/Styles/main.css in case of virtual directory.
For example in dev / QA environment that is perfect but for production I use a CDN for my assets so, would be very nice to have put the CDN domain into the configuration file and replace the Tilde with a custom value..

Basically something like this (Spark View Engine style)

``` xml
<resources>
      <add match="~/content/css" location="http://cdn.yourcdnprovider.com/youraccount/allstyles/css"/>
      <add match="~/content/js" location="http://cdn.yourcdnprovider.com/youraccount/appname/js"/>
</resources>
```

Does it make sense?
